src/main/java/core/basesyntax/Figure.java

```diff
@@ -0,0 +1,26 @@
+package core.basesyntax;
+
+public abstract class Figure implements Drawable {
+    private Color color;
+
+    public Figure(Color color) {
+        this.color = color;
+    }
+
+    public Color getColor() {
+        return color;
+    }
+
+    @Override
+    public void draw() {
+        System.out.println("Figure: " + getShape().getDescription().toLowerCase()
+                + ", area: " + String.format("%.2f", getArea()) + " sq.units, "
+                + getSizeDescription() + ", color: " + getColor().name().toLowerCase());
+    }
+
+    public abstract double getArea();
+
+    public abstract Shape getShape();
```

Comment:
I can't understand why name of the figure should be done via separate method.
Make it class property if you feel it necessary.
But for purposes it used here you may just call getClass().getSimpleName() method and you will get rather fine name of figure in draw() method without additional enum